service/internal/object/spec.go

```diff
@@ -0,0 +1,19 @@
+package object
+
+import "context"
+
+// Accessor defines an interface that can be implemented for different kinds of
+// objects to provide abstracted access to cluster details despite of undelying
+// types.
+type Accessor interface {
+	GetAPIEndpoint(ctx context.Context, obj interface{}) (string, error)
```

Comment:
So far only one method here to demonstrate usage for `tenantclients`, but my idea is to make this little bit similar to `key` package so that this interface can be used for abstracting the source of information.

First use case for this is to deal with different versions of Cluster API types while having the same handlers for both.